vpr/rr_graph.fbs

```diff
@@ -0,0 +1,176 @@
+namespace vtr.vpr;
+
+table ChannelList {
+  index:int;
+  info:int;
+}
+
+table Channels {
+  chan_width_max:int;
+  x_min:int;
+  y_min:int;
+  x_max:int;
+  y_max:int;
+  x_list:[ChannelList];
+  y_list:[ChannelList];
+}
+
+enum SwitchType : byte {
+  INVALID_SWITCH_TYPE,
+  MUX,
+  TRISTATE,
+  PASS_GATE,
+  SHORT,
+  BUFFER
+}
+
+table SwitchTiming {
+  r:float;
+  c_in:float;
+  c_out:float;
+  t_del:float;
+}
+
+table SwitchSizing {
+  mux_trans_size:float;
+  buf_size:float;
+}
+
+table Switch {
+  id:int;
+  name:string;
+  type:SwitchType;
+  timing:SwitchTiming;
+  sizing:SwitchSizing;
+}
+
+table SegmentTiming {
+  r_per_meter:float;
+  c_per_meter:float;
+}
+
+table Segment {
+  id:int;
+  name:string;
+  timing:SegmentTiming;
+}
+
+enum PinType : byte {
+  NONE,
+  OPEN,
+  OUTPUT,
+  INPUT
+}
+
+table Pin {
+  ptc:int;
+  name:string;
+}
+
+table PinClass {
+  type:PinType;
+  pin:[Pin];
+}
+
+table BlockType {
+  id:int;
+  name:string;
+  width:int;
+  height:int;
+  pin_class:[PinClass];
+}
+
+table GridLoc {
+  x:int;
+  y:int;
+  block_type_id:int;
+  width_offset:int;
+  height_offset:int;
+}
+
+enum NodeType: byte {
+  INVALID_NODE_TYPE,
+  CHANX,
```

Comment:
Probably really only need `CHAN`, it's a `CHANX` if xlow == xhigh and `CHANY` if ylow == yhigh.